server.py

```diff
@@ -83,21 +83,33 @@
         async for msg in ws:
             message = msg.json(loads=json.loads)
             if "interface_data" in message:
-                robot.power_down = message["interface_data"]["power_down"]
-                robot.program = []
-                for card_index in message["interface_data"]["my_program"]:
-                    if card_index is None:
-                        robot.program.append(None)
-                    else:
-                        robot.program.append(dealt_cards[card_index])
-
-                print(robot.program)
-                print(message)
-                # Send messages to all connected clients
-                ws_all = ws_receivers + ws_interfaces
-                for client in ws_all:
+                # it is still possible to choose cards
+                # TODO: not only by pressing key but also with time up
+                if not message["interface_data"]["confirmed"]:
```

Comment:
A tohle celé vůbec nechápu, proč tady je to `if not` asi si to přečtu desetkrát a napsat co to vlastně má udělat.
To `confirmed`tam je přece vždy, ať už True nebo False. Nejde mi to do hlavy co ta podmínka říká.

Reply:
Už mi to začíná docházet snad. Takže dokud není confirmed True, tak se přiřadí power_down, a robot_program... 
Jinak obecně mi vadí výraz `used_card_pack`, to jsou ty odhozené karty že? I v tom printu tady dole máš ` print("used", state.used_cards_pack)` ale já když vidím to USED tak to pro mě jsou karty, které jsem použila, vybrala, a ne co jsem odhodila, nepoužila.

Reply:
Jaký máš jiný návrh? Zatím máme new_cards_pack, tedy balíček karet, které jsme ještě neviděli, a balíček karet použitých. K tomu druhému patří jak odhozené, tak zahrané karty (=všechny, které se už ve hře nějak odkryly). Hoď nějaké nápady, já mám v tom teď blok.

Reply:
`card_pack` prejmenuju na `deck`, ale ted jake jsou ty balicky? Nove/stare, stavajici/minule? Skyte/videne? Nic mi uplne nefunguje v tomto kontextu.

Reply:
new card deck je jasný, udělá se na začátku a je skrytý... všechny jsou skryté jako pro všechny ne? každý vidí jen své rozdané karty. A v tom kodu se někde používají v té druhé skupině ty dvě rozdělení - odhozené a zahrané? Nebo jdou rovnou jen do jednoho? Jako vhodný název...  první kravina mě napadla - old_cards :-D  Teď koukám žes to už přejmenovala na present a past... nechceš teda nechat new a old? :-D Ten první balíček se vytvoří vlastně jen jednou že? Tak klidně může být present... jako cokoli jen mi vadilo to used :)

Reply:
"old" mi zde zase nesedí. nejsou to staré karty. Přejmenování je otázka minuty, tak až se na něčem shodneme, udělám finální změnu. Stejně ta funkcionalita není ještě top, tak budu přidávat další změny.